cyberark/conjur-policy-generator
# Clamps numeric parameters in web UI

#### What does this PR do?

In the web UI, each modification of a numeric parameter is clamped. So if the minimum number of users is 1, and you click on "minus" when it's already at 1, nothing will happen.

#### What's the background on this

Previously, you could set the values to nonsensical values which could cause the UI to become unresponsive. Clamping will make the UI more robust against misclicks and aggressive exploration 😈 

#### How should a reviewer test this manually?

1. check out the `clamp-numeric-parameters` branch
2. run `bin/tests`
3. run `bin/dev` to set up a dev server
4. visit localhost:9292 and click "minus" on the numeric parameters on the bottom
5. observe that they will refuse to go below 1
